src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CoreCaseDataService.java

```diff
@@ -38,91 +50,117 @@
+    private final AuthTokenGenerator authTokenGenerator;
+    private final CaseAccessApi caseAccessApi;
 
     @SuppressWarnings("squid:S00107") // All parameters are required here
     @Autowired
     public CoreCaseDataService(
-        SaveCoreCaseDataService saveCoreCaseDataService,
-        UpdateCoreCaseDataService updateCoreCaseDataService,
         CaseMapper caseMapper,
         CountyCourtJudgmentMapper countyCourtJudgmentMapper,
         ResponseMapper responseMapper,
         SettlementMapper settlementMapper,
         UserService userService,
         JsonMapper jsonMapper,
-        ReferenceNumberService referenceNumberService
+        ReferenceNumberService referenceNumberService,
+        CoreCaseDataApi coreCaseDataApi,
+        AuthTokenGenerator authTokenGenerator,
+        CaseAccessApi caseAccessApi
     ) {
-        this.saveCoreCaseDataService = saveCoreCaseDataService;
-        this.updateCoreCaseDataService = updateCoreCaseDataService;
         this.caseMapper = caseMapper;
         this.countyCourtJudgmentMapper = countyCourtJudgmentMapper;
         this.responseMapper = responseMapper;
         this.settlementMapper = settlementMapper;
         this.userService = userService;
         this.jsonMapper = jsonMapper;
         this.referenceNumberService = referenceNumberService;
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.authTokenGenerator = authTokenGenerator;
+        this.caseAccessApi = caseAccessApi;
     }
 
-    public Claim save(String authorisation, Claim claim) {
-        boolean claimantRepresented = claim.getClaimData().isClaimantRepresented();
-        String referenceNumber = referenceNumberService.getReferenceNumber(claimantRepresented);
+    public CaseReference savePrePayment(String authorisation, String externalId) {
+        UserDetails user = userService.getUserDetails(authorisation);
+
         try {
-            CCDCase ccdCase = caseMapper.to(claim);
-            ccdCase.setReferenceNumber(referenceNumber);
+            Map<String, Object> data = new HashMap<>();
+            data.put("externalId", externalId);
             EventRequestData eventRequestData = EventRequestData.builder()
-                .userId(claim.getSubmitterId())
+                .userId(user.getId())
                 .jurisdictionId(JURISDICTION_ID)
                 .caseTypeId(CASE_TYPE_ID)
-                .eventId(SUBMIT_CLAIM.getValue())
+                .eventId(SUBMIT_PRE_PAYMENT.getValue())
                 .ignoreWarning(true)
                 .build();
 
-            CaseDetails caseDetails = saveCoreCaseDataService
-                .save(
-                    authorisation,
-                    eventRequestData,
-                    ccdCase,
-                    claimantRepresented,
-                    claim.getLetterHolderId()
-                );
+            StartEventResponse startEventResponse = startCreate(authorisation, eventRequestData, user.isSolicitor());
```

Comment:
I think this needs to do a search before creating, what happens if the payment fails and then this call gets made again?
it would create 2 on hold cases and then save would fail as it finds multiple